lizmap/modules/lizmap/classes/qgisExpressionUtils.class.php

```diff
@@ -0,0 +1,282 @@
+<?php
+/**
+ * @author    3liz
+ * @copyright 2020 3liz
+ *
+ * @see      http://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+class qgisExpressionUtils
+{
+
+    /**
+     * Return criteria dependencies (fields) for a QGIS expression
+     *
+     * @param string $exp A QGIS expression string
+     *
+     * @return array The list of criteria dependencies
+     *
+     */
+    static public function getCriteriaFromExpression($exp)
+    {
+        if ($exp === null || $exp === '') {
+            return array();
+        }
+        preg_match_all('/"([^"]+)"/', $exp, $matches);
```

Comment:
We will have to write in doc that fields must **always** be defined with double-quote like `"my_field" > 10`. It is not mandatory in QGIS expression builder, so people could write valid expressions like `my_field > 10`

Reply:
Ah yes, indeed.
Can we raise a warning in the logs or a hint some where on execution time?

Reply:
needs doc ;-)